relay-profiling/src/sample/v2.rs

```diff
@@ -99,6 +101,12 @@ pub struct ProfileData {
 }
 
 impl ProfileData {
+    fn is_above_max_duration(&self) -> bool {
+        self.samples.last().is_some_and(|last_sample| {
+            let duration = last_sample.timestamp - self.samples[0].timestamp;
```

Comment:
This implies it's already sorted. It should be, but I think we should look for `min`/`max` here instead.